src/py42/clients/orgs.py

```diff
@@ -1,8 +1,224 @@
+            u"value": bool_to_str(value),
+            u"locked": False,
+        }
+        if not value:  # disable everything but FMC, like the UI does
+            self.endpoint_monitoring_removable_media_enabled = False
+            self.endpoint_monitoring_browser_and_applications_enabled = False
+            self.endpoint_monitoring_cloud_sync_enabled = False
+
+    @property
+    def endpoint_monitoring_removable_media_enabled(self):
+        value = self._t_settings[u"org-securityTools-device-detection-enable"][u"value"]
+        return str_to_bool(value)
+
+    @endpoint_monitoring_removable_media_enabled.setter
+    @bool_required
+    def endpoint_monitoring_removable_media_enabled(self, value):
+        if value:
+            self.endpoint_monitoring_enabled = True
+        self._t_settings[u"org-securityTools-device-detection-enable"] = {
+            u"key": u"org-securityTools-device-detection-enable",
+            u"value": bool_to_str(value),
+            u"locked": False,
+        }
+
+    @property
+    def endpoint_monitoring_cloud_sync_enabled(self):
+        value = self._t_settings[u"org-securityTools-cloud-detection-enable"][u"value"]
+        return str_to_bool(value)
+
+    @endpoint_monitoring_cloud_sync_enabled.setter
+    @bool_required
+    def endpoint_monitoring_cloud_sync_enabled(self, value):
+        if value:
+            self.endpoint_monitoring_enabled = True
+        self._t_settings[u"org-securityTools-cloud-detection-enable"] = {
+            u"key": u"org-securityTools-cloud-detection-enable",
+            u"value": bool_to_str(value),
+            u"locked": False,
+        }
+
+    @property
+    def endpoint_monitoring_browser_and_applications_enabled(self):
+        value = self._t_settings[u"org-securityTools-open-file-detection-enable"][
+            u"value"
+        ]
+        return str_to_bool(value)
+
+    @endpoint_monitoring_browser_and_applications_enabled.setter
+    @bool_required
+    def endpoint_monitoring_browser_and_applications_enabled(self, value):
+        if value:
+            self.endpoint_monitoring_enabled = True
+        self._t_settings[u"org-securityTools-open-file-detection-enable"] = {
+            u"key": u"org-securityTools-open-file-detection-enable",
+            u"value": bool_to_str(value),
+            u"locked": False,
+        }
+
+    @property
+    def endpoint_monitoring_file_metadata_enabled(self):
+        value = self._t_settings[u"device_fileForensics_enabled"][u"value"]
+        return str_to_bool(value)
+
+    @endpoint_monitoring_file_metadata_enabled.setter
+    @bool_required
+    def endpoint_monitoring_file_metadata_enabled(self, value):
+        if value:
+            self.endpoint_monitoring_enabled = True
+        self._t_settings[u"device_fileForensics_enabled"] = {
+            u"key": u"device_fileForensics_enabled",
+            u"value": bool_to_str(value),
+            u"locked": False,
+        }
+
+    def update(self):
```

Comment:
To avoid state-related issues, I wonder if we should separate the update command from the structure that is managing all of the state of the settings. Sort of like how we work with queries for alerts/files.
```python
class SettingsManager:
     def propx
     def propy
     def etc

class OrgService:
     def update_settings(settings_manager):
```

Reply:
I'm not opposed to making the update function external to the manager and having it take a manager as an argument, but I'm not sure what state-related problems that might solve.

Reply:
The "problem" is more of a usability one -- if `update` and the fields are attached to the same object, should I expect that when update is called that all of the properties that I have set have been cleared or not? If the two are separated I think the expectation that the properties do _not_ get cleared is much higher.

Reply:
I'm not quite sure what you're meaning by "cleared" here. 

Changes to the properties ultimately change the underlying dict that will be the body to be posted, which already exists with the _current_ org state before the user makes any changes. So I can check `orgsetting.org_name` and see that it's "MyOrg", then update the org name by setting the property `orgsetting.org_name = "MyNewOrg"`, after calling `update()` you'd expect it to go back to "MyOrg"? Or are you just talking about the manager class's state knowledge of what's been changed? So after update, all the properties reflect what the org currently is now, but now there's no values in the `.changes` property?

Reply:
The latter is closer to what I referring to. Imagine this scenario:
```python
# make a bunch of changes to the settings manager

settings_mgr.update()

# if it worked, do some other logic that helps me decide if I should change
# some additional settings

settings_mgr.some_other_settings = true

settings_mgr.upate()

# were all of the previous properties that I sent the first time I called `update()` resubmitted or not?
# did `update()` "unstage" the changes that I made successfully earlier?
```

The outcome becomes a little more clear (at least to in my opinion) when we separate things a bit and just treat the settings manager as a typical dict / struct:
```python
# make a bunch of changes to the settings manager

org_client.update_settings(settings_mgr)

# if it worked, do some other logic that helps me decide if I should change
# additional settings

settings_mgr.some_other_setting = true

# oops, I'm using the same object as I was before here, so this will _probably_ result in me
# re-sending the previous properties...
org_client.upate_settings(settings_mgr)

settings_mgr = SettingsMgr()
settings.new_setting = True

# now I know for sure that only new_setting will be sent
org_client.update_settings(settings_mgr)
```

Reply:
I understand that both approaches will have the same result regardless of whether or not things are "unstaged" in the first scenario. My point is that users do not understand that this an abstraction over submitting the entire current org state, and could be less confident in what the outcome of what theyre running will be in that case.

Reply:
I'm probably splitting hairs a little bit here and I'm not going to take an aggressive stand on this -- it's a thought, not a requirement, and this implementation does what it's supposed to do well.